addons/website_sale/controllers/main.py

```diff
@@ -1249,6 +1249,10 @@ def shop_payment_transaction(self, order_id, access_token, **kwargs):
         except AccessError:
             raise ValidationError("The access token is invalid.")
 
+        order = request.env['sale.order'].browse(order_id)
```

Comment:
Test for the validity of the id first, so it crashes ASAP if it needs to. This way we're certain to read the actual state of the order.
```suggestion
        order = request.env['sale.order'].browse(order_id).exists()
```

Reply:
The access check above already checks for the records existence so this isn't really necessary